test/index.test.mjs

```diff
@@ -0,0 +1,170 @@
+		expect(html`<a href="/hello" />`).toEqual({ tag: 'a', props: { href: '/hello' }, children: [] });
+	});
+	
+	test('single prop with static value followed by a single boolean prop', () => {
+		expect(html`<a href="/hello" b />`).toEqual({ tag: 'a', props: { href: '/hello', b: true }, children: [] });
+	});
+
+	test('two props with static values', () => {
+		expect(html`<a href="/hello" target="_blank" />`).toEqual({ tag: 'a', props: { href: '/hello', target: '_blank' }, children: [] });
+	});
+
+	test('single prop with dynamic value', () => {
+		expect(html`<a href=${'foo'} />`).toEqual({ tag: 'a', props: { href: 'foo' }, children: [] });
+	});
+
+	test('two props with dynamic values', () => {
+		function onClick(e) { }
+		expect(html`<a href=${'foo'} onClick=${onClick} />`).toEqual({ tag: 'a', props: { href: 'foo', onClick }, children: [] });
+	});
+
+	test('prop with quoted dynamic value ignores static parts', () => {
+		expect(html`<a href="before${'foo'}after" a="b" />`).toEqual({ tag: 'a', props: { href: 'foo', a: 'b' }, children: [] });
+	});
+
+	test('spread props', () => {
+		expect(html`<a ...${{ foo: 'bar' }} />`).toEqual({ tag: 'a', props: { foo: 'bar' }, children: [] });
+		expect(html`<a b ...${{ foo: 'bar' }} />`).toEqual({ tag: 'a', props: { b: true, foo: 'bar' }, children: [] });
+		expect(html`<a b c ...${{ foo: 'bar' }} />`).toEqual({ tag: 'a', props: { b: true, c: true, foo: 'bar' }, children: [] });
+		expect(html`<a ...${{ foo: 'bar' }} b />`).toEqual({ tag: 'a', props: { b: true, foo: 'bar' }, children: [] });
+		expect(html`<a b="1" ...${{ foo: 'bar' }} />`).toEqual({ tag: 'a', props: { b: '1', foo: 'bar' }, children: [] });
+		expect(html`<a x="1"><b y="2" ...${{ c: 'bar' }}/></a>`).toEqual(h('a', { x: '1' }, h('b', { y: '2', c: 'bar' }) ));
+		expect(html`<a ...${{ c: 'bar' }}><b ...${{ d: 'baz' }}/></a>`).toEqual(h('a', { c: 'bar' }, h('b', { d: 'baz' }) ));
+	});
+
+	test('multiple spread props in one element', () => {
+		expect(html`<a ...${{ foo: 'bar' }} ...${{ quux: 'baz' }} />`).toEqual({ tag: 'a', props: { foo: 'bar', quux: 'baz' }, children: [] });
+	});
+  
+	test('mixed spread + static props', () => {
+		expect(html`<a b ...${{ foo: 'bar' }} />`).toEqual({ tag: 'a', props: { b: true, foo: 'bar' }, children: [] });
+		expect(html`<a b c ...${{ foo: 'bar' }} />`).toEqual({ tag: 'a', props: { b: true, c: true, foo: 'bar' }, children: [] });
+		expect(html`<a ...${{ foo: 'bar' }} b />`).toEqual({ tag: 'a', props: { b: true, foo: 'bar' }, children: [] });
+		expect(html`<a ...${{ foo: 'bar' }} b c />`).toEqual({ tag: 'a', props: { b: true, c: true, foo: 'bar' }, children: [] });
+	});
+
+	test('closing tag', () => {
+		expect(html`<a></a>`).toEqual({ tag: 'a', props: null, children: [] });
+		expect(html`<a b></a>`).toEqual({ tag: 'a', props: { b: true }, children: [] });
+	});
+
+	test('auto-closing tag', () => {
+		expect(html`<a><//>`).toEqual({ tag: 'a', props: null, children: [] });
```

Comment:
Not related to the PR - but what was the motivation for double-slash closing tag? JSX introduces `</>` for the purpose of closing fragments, and tbh I had that expectation here too. Are there some collisions?

Reply:
The original reason was that it was easy to parse - end tag names were ignored in the HTML parser version and are ignored in the custom parser version too. Really you could give the tag any name, like `<a></>`.

Reply:
Does the new parser ignore end tags due to the same reason?

Reply:
just to save bytes. It doesn't validate the opening and closing tags match.